File: day-1/part-2/miles.exs
Code.require_file("miles_input.exs", "./")

defmodule Part2 do
  def sum_list([head | tail], sum, sum_counts) do
    if found_frequency_enough?(Map.get(sum_counts, sum)) do
      sum
    else
      sum_list(
        tail,
        sum + head,
        Map.put(sum_counts, sum + head, Map.get(sum_counts, sum + head, 0) + 1)
      )
    end
  end

  def sum_list([], sum, list) do
    sum_list(In.put, sum, list)
  end

  defp found_frequency_enough?(2), do: true
  defp found_frequency_enough?(_), do: false
end

Part2.sum_list(In.put, 0, %{})
|> IO.puts

File: day-1/part-2/miles.rb
require_relative "./miles_input"

sum_counts = Hash.new(0)
sum = 0
counter = 0
while true
  input = INPUT[counter]
  sum_counts[sum] += 1
  break(sum) if sum_counts[sum] > 1
  sum += input
  counter += 1
  counter = 0 if counter >= INPUT.size
end

puts sum
